pkgs/development/libraries/onnxruntime/default.nix

```diff
@@ -0,0 +1,115 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchpatch
+, fetchurl
+, pkg-config
+, cmake
+, python3
+, libpng
+, zlib
+, eigen
+, protobuf
+, howard-hinnant-date
+, nlohmann_json
+, boost
+, oneDNN
+, gtest
+}:
+
+let
+  abseil = fetchurl {
+    url = "https://github.com/abseil/abseil-cpp/archive/refs/tags/20211102.0.zip";
+    sha1 = "ce61532df974d00025b1220408ce1c900d81baf2";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "onnxruntime";
+  version = "1.12.1";
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "onnxruntime";
+    rev = "v${version}";
+    sha256 = "sha256-wwllEemiHTp9aJcCd1gsTS4WUVMp5wW+4i/+6DzmAeM=";
+    fetchSubmodules = true;
+  };
+
+  patches = [
+    # Use dnnl from nixpkgs instead of submodules
+    (fetchpatch {
+      name = "system-dnnl.patch";
+      url = "https://aur.archlinux.org/cgit/aur.git/plain/system-dnnl.diff?h=python-onnxruntime&id=0185531906bda3a9aba93bbb0f3dcfeb0ae671ad";
+      sha256 = "sha256-58RBrQnAWNtc/1pmFs+PkZ6qCsL1LfMY3P0exMKzotA=";
+    })
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    python3
+  ];
+
+  buildInputs = [
+    libpng
+    zlib
+    protobuf
+    howard-hinnant-date
+    nlohmann_json
+    boost
+    oneDNN
+    gtest
```

Comment:
I wonder if this should be in `nativeBuildInputs` instead, the binaries generated with it should only get executed on the host compiling the package. It might break cross compilation otherwise, although I'm not entirely sure it's the case here.

You could test if cross compiling works with the following:
```shell
~/nixpkgs $ nix build -Lf . pkgsCross.aarch64-multiplatform.onnxruntime
```

Reply:
Moved gtest to native build inputs, but cross compilation for package oneDNN failed.